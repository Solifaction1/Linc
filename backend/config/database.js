const path = require('path');

const connection = process.env.NODE_ENV !== 'production' ?
  {
    client: 'sqlite',
    connection: {
      filename: path.join(__dirname, '..', '.tmp/data.db'),
    },
    useNullAsDefault: true,
  } : {
    client: "postgres",
    acquireConnectionTimeout: 120000,
    connection: {
      // host: `/cloudsql/${process.env.INSTANCE_CONNECTION_NAME}`, // for a PostgreSQL database
      host: process.env.DATABASE_HOST, // for a PostgreSQL database
      database: process.env.DATABASE_NAME,
      user: process.env.DATABASE_USERNAME,
      password: process.env.DATABASE_PASSWORD,
      ssl: false,

    },
    useNullAsDefault: true,
    debug: false,
  };

module.exports = ({ }) => ({
  connection: connection,
});
