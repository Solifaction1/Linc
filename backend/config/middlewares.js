module.exports = [
  'strapi::errors',
  {
    name: 'strapi::cors',
    config: {
      enabled: true,
      headers: ['*'],
      origin: ['*']
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            "cdn.jsdelivr.net",
            "strapi.io",
            "s3.amazonaws.com",
            "editor.unlayer.com",
            process.env.CDN_URL,
          ],
          "script-src": ["'self'", "editor.unlayer.com"],
          "frame-src": ["'self'", "editor.unlayer.com"],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            process.env.CDN_URL,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
];
