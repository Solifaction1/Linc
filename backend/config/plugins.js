module.exports = ({ env }) => ({
  "strapi-plugin-populate-deep": {
    config: {
      defaultDepth: 5, // Default is 5
    },
  },
  "duplicate-button": true,
  email: {
    config: {
      provider: "mailgun",
      providerOptions: {
        key: env("MAILGUN_API_KEY"), // Required
        domain: env("MAILGUN_DOMAIN"), // Required
        url: "https://api.eu.mailgun.net", //Optional. If domain region is Europe use 'https://api.eu.mailgun.net'
      },
      settings: {
        defaultFrom: "<mg@linc.legal>",
        defaultReplyTo: "<mg@linc.legal>",
      },
    },
  },
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
  "content-versioning": {
    enabled: false,
  },
  slugify: {
    enabled: false,
    config: {
      contentTypes: {
        blog: {
          field: "slug",
          references: "title",
        },
      },
    },
  },
  "preview-button": {
    enabled: true,
    resolve: "./src/plugins/preview-button", // path to plugin folder
  },
  "email-designer": {
    enabled: true,

    // ⬇︎ Add the config property
    config: {
      editor: {
        // optional - if you have a premium unlayer account
        // projectId: [UNLAYER_PROJECT_ID],

        tools: {
          heading: {
            properties: {
              text: {
                value: "This is the new default text!",
              },
            },
          },
        },
        options: {
          features: {
            colorPicker: {
              presets: ["#D9E3F0", "#F47373", "#697689", "#37D67A"],
            },
          },
          fonts: {
            showDefaultFonts: false,
            /*
             * If you want use a custom font you need a premium unlayer account and pass a projectId number :-(
             */
            customFonts: [
              {
                label: "Anton",
                value: "'Anton', sans-serif",
                url: "https://fonts.googleapis.com/css?family=Anton",
              },
              {
                label: "Lato",
                value: "'Lato', Tahoma, Verdana, sans-serif",
                url: "https://fonts.googleapis.com/css?family=Lato",
              },
              // ...
            ],
          },
          mergeTags: [
            {
              name: "Naam",
              value: "{{= USER.Naam }}",
              sample: "Naam",
            },
            {
              name: "Voornaam",
              value: "{{= USER.Voornaam }}",
              sample: "Voornaam",
            },
            {
              name: "Email",
              value: "{{= USER.Email }}",
              sample: "john@doe.com",
            },
            {
              name: "Telefoonnummer",
              value: "{{= USER.Telefoonnummer }}",
              sample: "0123456789",
            },
            {
              name: "select",
              value: "{{= USER.select }}",
              sample: "select",
            },
            {
              name: "textArea",
              value: "{{= USER.textArea }}",
              sample: "textArea",
            },
            {
              name: "Form ID",
              value: "{{= USER.formId }}",
              sample: "0",
            },
          ],
        },
        appearance: {
          theme: "dark",
          panels: {
            tools: {
              dock: "left",
            },
          },
        },
      },
    },
  },
  seo: {
    enabled: true,
  },
  ckeditor: {
    enabled: true,
    config: {
      plugin: {
        // disable data-theme tag setting //
        // setAttribute:false,
        // disable strapi theme, will use default ckeditor theme //
        // strapiTheme:false,
        // styles applied to editor container (global scope) //
        // styles:`
        // :root{
        //   --ck-color-focus-border:red;
        //   --ck-color-text:red;
        // }
        // `
        styles: `
        .ck-editor__main p,
        .ck.ck-list__item .ck-button .ck-button__label {
          line-height: normal;
        }`,
      },
      editor: {
        // editor default config

        // https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html
        // if you need markdown support and output set: removePlugins: [''],
        // default is
        removePlugins: ["Markdown"],

        // https://ckeditor.com/docs/ckeditor5/latest/features/toolbar/toolbar.html
        toolbar: {
          items: [
            "paragraph",
            "heading1",
            "heading2",
            "heading3",
            "|",
            "bold",
            "italic",
            "underline",
            "fontSize",
            "removeFormat",
            "|",
            "bulletedList",
            "numberedList",
            "alignment",
            "|",
            "StrapiMediaLib",
            "link",
            "horizontalLine",
            "|",
            "sourceEditing",
            "fullScreen",
            "undo",
            "redo",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/font.html
        fontSize: {
          options: [
            9,
            11,
            13,
            {
              title: "default",
              model: "18px",
            },
            17,
            19,
            21,
            27,
            35,
          ],
          supportAllValues: false,
        },
        fontFamily: {
          options: [
            "default",
            "Arial, Helvetica Neue, Helvetica, Source Sans Pro, sans-serif",
            "Courier New, Courier, monospace",
            "Georgia, serif",
            "Lucida Sans Unicode, Lucida Grande, sans-serif",
            "Tahoma, Geneva, sans-serif",
            "Times New Roman, Times, serif",
            "Trebuchet MS, Helvetica, sans-serif",
            "Verdana, Geneva, sans-serif",
            "Roboto, Roboto Black, Roboto Medium, Roboto Light, sans-serif",
          ],
          supportAllValues: true,
        },
        fontColor: {
          columns: 5,
          documentColors: 10,
        },
        fontBackgroundColor: {
          columns: 5,
          documentColors: 10,
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
        // default language: 'en',
        // https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html
        image: {
          resizeUnit: "%",
          resizeOptions: [
            {
              name: "resizeImage:original",
              value: null,
              icon: "original",
            },
            {
              name: "resizeImage:25",
              value: "25",
              icon: "small",
            },
            {
              name: "resizeImage:50",
              value: "50",
              icon: "medium",
            },
            {
              name: "resizeImage:75",
              value: "75",
              icon: "large",
            },
          ],
          toolbar: [
            "toggleImageCaption",
            "imageTextAlternative",
            "imageStyle:inline",
            "imageStyle:block",
            "imageStyle:side",
            "linkImage",
            "resizeImage:25",
            "resizeImage:50",
            "resizeImage:75",
            "resizeImage:original",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/table.html
        table: {
          contentToolbar: [
            "tableColumn",
            "tableRow",
            "mergeTableCells",
            "tableCellProperties",
            "tableProperties",
            "toggleTableCaption",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/headings.html
        heading: {
          options: [
            {
              model: "paragraph",
              title: "Paragraph",
              class: "ck-heading_paragraph",
            },
            {
              model: "heading1",
              view: "h1",
              title: "Heading 1",
              class: "ck-heading_heading1",
            },
            {
              model: "heading2",
              view: "h2",
              title: "Heading 2",
              class: "ck-heading_heading2",
            },
            {
              model: "heading3",
              view: "h3",
              title: "Heading 3",
              class: "ck-heading_heading3",
            },
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html
        htmlSupport: {
          allow: [
            {
              name: "img",
              attributes: {
                sizes: true,
                loading: true,
              },
            },
          ],
        },
      },
    },
  },
});
