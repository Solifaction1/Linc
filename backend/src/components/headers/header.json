{
  "collectionName": "components_headers_header",
  "info": {
    "displayName": "Header",
    "description": ""
  },
  "options": {},
  "attributes": {
    "text": {
      "type": "component",
      "repeatable": false,
      "component": "text.ygwys"
    },
    "image": {
      "type": "media",
      "multiple": false,
      "required": false,
      "allowedTypes": ["images"]
    },
    "CTA": {
      "type": "component",
      "repeatable": false,
      "component": "sections.cta"
    },
    "bottomText": {
      "type": "component",
      "repeatable": false,
      "component": "tiles.ygwys-cta-tile"
    },
    "imagePosition": {
      "type": "enumeration",
      "enum": ["left", "right"],
      "default": "right"
    }
  }
}
