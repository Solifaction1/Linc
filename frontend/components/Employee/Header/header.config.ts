import { EmployeeHeaderData } from '@models/attributes.model'

const defaultProps: EmployeeHeaderData = {
	color: 'gray',
	imagePosition: 'right',
	image: {
		data: {
			id: 0,
			attributes: {
				name: 'name',
				alternativeText: 'alternativeText',
				width: 876,
				height: 1025,
				hash: 'hash',
				ext: 'ext',
				mime: 'mime',
				size: 0,
				url: 'https://via.placeholder.com/876x1025',
			},
		},
	},
	text: {
		text: '<p>description</p>',
		className: 'string',
	},
	CTA: null,
	bottomText: {
		description: {
			text: '<p>description bottom</p>',
			className: 'string',
		},
		cta: null
	},
	firstName: 'firstName',
	name: 'name',
	telephone: '+123456789',
	rechtsdomeinen: {
		data: []
	}
}

export { defaultProps }
