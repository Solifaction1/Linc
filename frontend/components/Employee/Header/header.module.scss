@import 'styles/vars';
@import 'styles/utilities';

.header {
	padding-top: 7.5rem;

	.headerContainer {
		@extend %page-container;
		position: relative;
		z-index: 3;
		grid-template-areas: 'headertext headertext headertext headertext headertext . headerimg headerimg headerimg headerimg headerimg headerimg';

		padding-top: unset;
		padding-bottom: unset;

		@media (max-width: $break-lg) {
			grid-template-areas:
				'headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext'
				'headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg';

			margin-top: calc(#{$spacing-base} * 9);
			padding: calc(#{$spacing-base} * 4) $page-bleed;
			padding-bottom: unset;
		}

		@media (max-width: $break-s) {
			padding: calc(#{$spacing-base} * 8) $page-bleed-mobile;
		}

		&::after {
			content: '';
			position: absolute;
			right: calc(49% - 6.4rem);
			bottom: 0;
			z-index: 2;

			width: 7.57rem;
			height: 7.5rem;

			background: $color-gray;

			opacity: 0.9;

			mix-blend-mode: multiply;

			@media (max-width: $break-lg) {
				right: unset;
				bottom: 0;
				left: $page-bleed;

				width: 4.6rem;
				height: 4.6rem;
			}

			@media (max-width: $break-s) {
				left: $page-bleed-mobile;
			}
		}

		.headerTextWrapper {
			position: relative;
			grid-area: headertext;
			padding-bottom: 7.49rem;

			.text {
				h1 {
					@extend %h1;
				}

				p, a {
					@extend %body-copy-1;
				}

				p:not(.title), .title:first-of-type {
					margin-top: calc(#{$spacing-base} * 4);
				}

				.title {
					font-weight: $font-weight-bold;
				}
			}

			button {
				margin-top: calc(#{$spacing-base} * 4);
			}
		}

		.headerImageWrapper {
			position: absolute;
			grid-area: headerimg;
			width: 50vw;

			@media (max-width: $break-lg) {
				position: relative;
				display: block;

				width: 100%;
				height: 51.5rem;
				margin-bottom: -19.5rem;
			}

			@media (max-width: $break-s) {
				top: 5.8rem;
				margin-bottom: -10.5rem;
			}

			img {
				object-fit: cover;

				width: 100%;
				max-width: 72rem;
				height: 100%;

				object-position: left;

				@media (max-width: $break-lg) {
					width: calc(100% + calc(#{$spacing-base} * 4.5));
					max-width: 100rem;
					height: 100%;

					object-position: top;
				}

				@media (max-width: $break-md) {
					height: 100%;
				}

				@media (max-width: $break-s) {
					width: calc(100% + calc(#{$spacing-base} * 3));
				}
			}
		}
	}

	.headerBottomContainer {
		margin-bottom: -7.5rem;
		padding-top: 12rem;
		background: white;

		@media (max-width: $break-lg) {
			margin-bottom: -3.5rem;
			padding-top: 28rem;
		}

		@media (max-width: $break-md) {
			margin-bottom: -3.5rem;
			padding-top: 28rem;
		}

		@media (max-width: $break-s) {
			margin-bottom: -3.5rem;
			padding-top: 20rem;
		}

		.bottomText {
			@extend %page-container;
			grid-template-areas: 'text text text text text . . . . . . .';
			padding-top: unset;

			@media (max-width: $break-lg) {
				grid-template-areas: 'text text text text text text text text text text text text';
			}

			.bottomTextWrapper {
				grid-area: text;

				.text {
					display: flex;
					flex-direction: column;
					color: $color-beige;

					h2 {
						@extend %h2;
						color: $color-beige-dark;
					}

					p {
						@extend %body-copy-1;
						margin-top: calc(#{$spacing-base} * 4);
						color: $color-gray;
					}
				}

				button {
					margin-top: calc(#{$spacing-base} * 4);
					border: 1px solid $color-beige-dark;

					p {
						color: $color-beige-dark;
					}
				}
			}
		}
	}

	// ---SHORT HEADER

	&.noBottomText {
		margin-bottom: calc(#{$spacing-base} * 19.3);

		.headerContainer .headerImageWrapper {
			height: calc(100% + 7.5rem);

			@media (max-width: $break-lg) {
				height: calc(100% + 2.6rem);
				height: 45.4rem;
				margin-bottom: -2.2rem;
			}

			@media (max-width: $break-md) {
				top: 0;
				height: 35rem;
			}

			@media (max-width: $break-s) {
				top: -2.2rem;
				margin-bottom: -10.5rem;
			}

			img {
				height: 100%;
				object-position: center;

				@media (max-width: $break-lg) {
					height: 100%;
				}

				@media (max-width: $break-md) {
					height: 100%;
					max-height: unset;
				}
			}
		}
	}

	// ---IMAGE POSITION

	&.right {
		.headerImageWrapper {
			@media (max-width: $break-hd) {
				text-align: right;
			}
		}
	}

	&.left {
		.headerContainer {
			grid-template-areas: 'headerimg headerimg headerimg headerimg headerimg headerimg . headertext headertext headertext headertext headertext ';
			padding-top: unset;
			padding-bottom: unset;

			@media (max-width: $break-lg) {
				grid-template-areas:
					'headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext'
					'headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg';
			}

			&::after {
				right: unset;
				left: calc(49% - 6.4rem);

				@media (max-width: $break-lg) {
					right: $page-bleed;
					bottom: 0;
					left: unset;

					width: 4.6rem;
					height: 4.6rem;
				}
			}

			.headerImageWrapper {
				right: calc(#{$spacing-base} * -0);

				@media (min-width: $break-hd) {
					text-align: right;
				}

				@media (max-width: $break-lg) {
					position: relative;
					display: block;

					width: 100%;
					margin-bottom: -21.5rem;
				}

				@media (max-width: $break-md) {
					top: -2.2rem;
				}

				@media (max-width: $break-s) {
					margin-bottom: -12.5rem;
				}

				img {
					@media (max-width: $break-lg) {
						position: relative;
						left: calc(0.8rem * (-4.5));
					}
				}
			}
		}

		.headerBottomContainer {
			.bottomText {
				grid-template-areas: '. . . . . . . text text text text text';

				@media (max-width: $break-lg) {
					grid-template-areas: 'text text text text text text text text text text text text';
				}
			}
		}
	}

	&.left.noBottomText {
		.headerContainer {
			.headerImageWrapper {
				@media (max-width: $break-lg) {
					top: -1.6rem;
					margin-bottom: -3.6rem;
				}

				@media (max-width: $break-md) {
					height: 35rem;
					margin-bottom: -4.6rem;
				}
			}
		}
	}

	// ---BACKGROUND COLORS

	&.beige {
		background-color: $color-beige;

		.headerContainer::after {
			background: $color-beige;
		}

		.headerTextWrapper .text h1,
		.headerTextWrapper .text h2,
		.headerTextWrapper .text h3,
		.headerTextWrapper .text li,
		.headerTextWrapper .text ul,
		.headerTextWrapper .text ol,
		.headerTextWrapper .text p,
		.headerTextWrapper .text a {
			color: $color-gray;
		}
	}

	&.blueGray {
		background-color: $color-blue-gray;

		.headerContainer::after {
			background: $color-blue-gray;
		}

		.headerTextWrapper .text h1,
		.headerTextWrapper .text h2,
		.headerTextWrapper .text h3,
		.headerTextWrapper .text li,
		.headerTextWrapper .text ul,
		.headerTextWrapper .text ol,
		.headerTextWrapper .text p,
		.headerTextWrapper .text a {
			color: $color-gray;
		}
	}

	&.darkGreen {
		background-color: $color-green-dark;

		.headerContainer::after {
			background: $color-green-dark;
		}

		.headerTextWrapper .text h1,
		.headerTextWrapper .text h2,
		.headerTextWrapper .text h3,
		.headerTextWrapper .text li,
		.headerTextWrapper .text ul,
		.headerTextWrapper .text ol,
		.headerTextWrapper .text p,
		.headerTextWrapper .text a {
			color: $color-beige;
		}
	}

	&.lightGreen {
		background-color: $color-green;

		.headerContainer::after {
			background: $color-green;
		}

		.headerTextWrapper .text h1,
		.headerTextWrapper .text h2,
		.headerTextWrapper .text h3,
		.headerTextWrapper .text li,
		.headerTextWrapper .text ul,
		.headerTextWrapper .text ol,
		.headerTextWrapper .text p,
		.headerTextWrapper .text a {
			color: $color-beige;
		}
	}

	&.gray {
		background-color: $color-gray;

		.headerContainer::after {
			background: $color-gray;
		}

		.headerTextWrapper .text h1,
		.headerTextWrapper .text h2,
		.headerTextWrapper .text h3,
		.headerTextWrapper .text li,
		.headerTextWrapper .text ul,
		.headerTextWrapper .text ol,
		.headerTextWrapper .text p,
		.headerTextWrapper .text a {
			color: $color-beige;
		}
	}
}
