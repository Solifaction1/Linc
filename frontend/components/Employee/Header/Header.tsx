import React from 'react'
import Image from 'next/image'
import { getUniqueId } from '@utils/utils.service'

import { EmployeeHeaderData } from '@models/attributes.model'
import SwitchCTA from '@templates/SwitchCTA'
import YGWYS from '@components/YGWYS/YGWYS'
import { getColor, getColorHeader } from '@constants/color'
import styles from './header.module.scss'
import { defaultProps } from './header.config'

const EmployeeHeader = (props: EmployeeHeaderData) => {
	const image = props.image?.data?.attributes
	return (
		<header
			className={`${styles.header} ${styles[getColor(props.color)]} ${
				styles[props.imagePosition]
			} ${
				props.bottomText?.description?.text
					? styles.bottomText
					: styles.noBottomText
			}`}
		>
			<div className={styles.headerContainer}>
				<div className={styles.headerTextWrapper}>
					{props.text?.text && (
						<div className={styles.text}>
							<h1>{`${`${props.firstName}`}${` ${props.name}`}`}</h1>

							{props.rechtsdomeinen?.data.map((rechtsdomein) => (
								<p key={rechtsdomein.id} className={styles.title}>
									{rechtsdomein.attributes.title}
								</p>
							))}

							<div className={styles.text}>
								<YGWYS text={props.text?.text} />
							</div>

							<a
								className={styles.text}
								href={`tel:${props.telephone}`}
							>{`Tel: ${props.telephone}`}</a>
						</div>
					)}

					{props.CTA && (
						<SwitchCTA
							ctaData={props.CTA}
							color={`${getColorHeader(props.color)}`}
						/>
					)}
				</div>
				<div className={styles.headerImageWrapper}>
					{image?.url && (
						<Image
							className={styles.image}
							key={getUniqueId()}
							width={image.width}
							height={image.height}
							src={image.url ?? ''}
							alt={image.alternativeText ?? ''}
							priority
							loader={() => {
								return `/api/custom-loader?url=${encodeURIComponent(
									image.url
								)}?q=${100}`
							}}
						/>
					)}
				</div>
			</div>
			{props.bottomText?.description?.text && (
				<div className={styles.headerBottomContainer}>
					<div className={styles.bottomText}>
						<div className={styles.bottomTextWrapper}>
							{props.bottomText?.description.text && (
								<div className={styles.text}>
									<YGWYS text={props.bottomText?.description.text} />
								</div>
							)}

							{props.bottomText.cta && (
								<SwitchCTA ctaData={props.bottomText.cta} />
							)}
						</div>
					</div>
				</div>
			)}
		</header>
	)
}

EmployeeHeader.defaultProps = defaultProps

export default EmployeeHeader
