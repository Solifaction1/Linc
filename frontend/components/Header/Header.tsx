import React from 'react'
import Image from 'next/image'
import { getUniqueId } from '@utils/utils.service'

import { HeaderData } from '@models/attributes.model'
import SwitchCTA from '@templates/SwitchCTA'
import YGWYS from '@components/YGWYS/YGWYS'
import { getColor, getColorHeader } from '@constants/color'
import styles from './header.module.scss'
import { defaultProps } from './header.config'

const Header = (props: HeaderData) => {
	const image = props.image?.data?.attributes
	return (
		<header
			className={`${styles.header} ${styles[getColor(props.color)]} ${
				styles[props.imagePosition]
			} ${
				props.bottomText?.description?.text
					? styles.bottomText
					: styles.noBottomText
			}
			${!image && styles.noImage}`}
		>
			<div className={styles.headerContainer}>
				<div className={styles.headerTextWrapper}>
					{props.text?.text && (
						<div className={styles.text}>
							<YGWYS text={props.text?.text} />
						</div>
					)}

					{props.CTA && (
						<SwitchCTA
							ctaData={props.CTA}
							color={`${getColorHeader(props.color)}`}
						/>
					)}
				</div>

				{image?.url && (
					<div className={styles.headerImageWrapper}>
						<Image
							className={styles.image}
							key={getUniqueId()}
							width={image.width}
							height={image.height}
							src={image.url ?? ''}
							alt={image.alternativeText ?? ''}
							priority
							loader={() => {
								return `/api/custom-loader?url=${encodeURIComponent(
									image.url
								)}?q=${100}`
							}}
						/>
					</div>
				)}
			</div>
			{props.bottomText?.description?.text && (
				<div className={styles.headerBottomContainer}>
					<div className={styles.bottomText}>
						<div className={styles.bottomTextWrapper}>
							{props.bottomText?.description.text && (
								<div className={styles.text}>
									<YGWYS text={props.bottomText?.description.text} />
								</div>
							)}

							{props.bottomText.cta && (
								<SwitchCTA ctaData={props.bottomText.cta} />
							)}
						</div>
					</div>
				</div>
			)}
		</header>
	)
}

Header.defaultProps = defaultProps

export default Header
