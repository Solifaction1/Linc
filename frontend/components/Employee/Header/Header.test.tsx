import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Header from './Header'
import { defaultProps } from './header.config'

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}))

describe('Header', () => {
	it('renders a Header', () => {
		const { container, getByText } = render(<Header {...defaultProps} />)

		expect(container).toMatchSnapshot()
		const header = container.querySelector('header')
		expect(header).toBeVisible()
		expect(header?.getAttribute('class')).toEqual(
			'header gray right bottomText'
		)

		expect(container.getElementsByClassName('headerContainer')[0]).toBeVisible()
		expect(
			container.getElementsByClassName('headerTextWrapper')[0]
		).toBeVisible()

		expect(container.getElementsByClassName('text')[0]).toBeVisible()
		expect(container.getElementsByClassName('ygwys')[0]).toBeVisible()
		expect(getByText('description', { selector: 'p' })).toBeVisible()

		expect(
			container.getElementsByClassName('headerImageWrapper')[0]
		).toBeVisible()
		expect(container.getElementsByClassName('image')[0]).toBeVisible()
		const img = container.querySelector('img')
		expect(img).toBeVisible()
		expect(img?.getAttribute('alt')).toEqual('alternativeText')
		expect(img?.getAttribute('width')).toEqual('876')
		expect(img?.getAttribute('height')).toEqual('1025')
		expect(img?.getAttribute('src')).toEqual(
			'/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F876x1025?q=100'
		)

		expect(
			container.getElementsByClassName('headerBottomContainer')[0]
		).toBeVisible()
		expect(container.getElementsByClassName('bottomText')[0]).toBeVisible()

		expect(
			container.getElementsByClassName('bottomTextWrapper')[0]
		).toBeVisible()
		expect(container.getElementsByClassName('text')[1]).toBeVisible()
		expect(container.getElementsByClassName('ygwys')[1]).toBeVisible()
		expect(getByText('description bottom', { selector: 'p' })).toBeVisible()
	})
})
