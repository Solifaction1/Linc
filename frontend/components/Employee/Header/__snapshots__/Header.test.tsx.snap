// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders a Header 1`] = `
<div>
  <header
    class="header gray right bottomText"
  >
    <div
      class="headerContainer"
    >
      <div
        class="headerTextWrapper"
      >
        <div
          class="text"
        >
          <h1>
            firstName name
          </h1>
          <div
            class="text"
          >
            <div
              class="ygwys 
			undefined undefined"
            >
              <p>
                description
              </p>
            </div>
          </div>
          <a
            class="text"
            href="tel:+123456789"
          >
            Tel: +123456789
          </a>
        </div>
      </div>
      <div
        class="headerImageWrapper"
      >
        <img
          alt="alternativeText"
          class="image"
          data-nimg="1"
          decoding="async"
          height="1025"
          src="/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F876x1025?q=100"
          srcset="/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F876x1025?q=100 1x, /api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F876x1025?q=100 2x"
          style="color: transparent;"
          width="876"
        />
      </div>
    </div>
    <div
      class="headerBottomContainer"
    >
      <div
        class="bottomText"
      >
        <div
          class="bottomTextWrapper"
        >
          <div
            class="text"
          >
            <div
              class="ygwys 
			undefined undefined"
            >
              <p>
                description bottom
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
